### PR TITLE
Disable distributed nccl tests to unblock Amazon2023 ami upgrade

### PR DESCRIPTION
**Pull request by ZainRizvi**
These tests keep failing on the Linux Amazon 2023 AMI.  The distributed team is looking into them, but until then, disabling the tests in order to unblock the AMI upgrade

Examples of the failures:
Failure 1: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963175
```
FAILED [90.0880s] distributed/test_c10d_nccl.py::NCCLTraceTestDumpOnTimeout::test_timeout_dumps_timing_enabled_False - AssertionError: None mismatch: None is not -6
```

Failure 2: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963494
```
____ NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks ____
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/distributed/test_c10d_nccl.py", line 4214, in test_timeout_dumps_on_stuck_ranks
    self.assertEqual(self._wait_process(0, timeout=90), -6)
  File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3721, in assertEqual
    raise error_metas.pop()[0].to_error(
AssertionError: None mismatch: None is not -6
```

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Comment by fduwjj** [comment]
where can we see the test failure? I can help with debugging these tests.

File: test/distributed/test_c10d_nccl.py [hunk 1]
```diff
@@ -4137,6 +4137,7 @@ def _wait_process(self, rank, timeout):
             return None
 
 
+@skip_but_pass_in_sandcastle
```
Comment (wconstab) [inline test/distributed/test_c10d_nccl.py:4140]: iiuc this macro would only skip in sandcastle, not oss CI?  or am i wrong

Comment (kit1980) [reply test/distributed/test_c10d_nccl.py:4140]: This will skip on GitHub and pass on sandcastle.

**Comment by ZainRizvi** [comment]
> where can we see the test failure? I can help with debugging these tests.

@c-p-i-o  is looking into the test failure. 

Here's an example of what they look like though:


Examples of the failures:
Failure 1: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963175
```
FAILED [90.0880s] distributed/test_c10d_nccl.py::NCCLTraceTestDumpOnTimeout::test_timeout_dumps_timing_enabled_False - AssertionError: None mismatch: None is not -6
```

Failure 2: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963494
```
____ NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks ____
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/distributed/test_c10d_nccl.py", line 4214, in test_timeout_dumps_on_stuck_ranks
    self.assertEqual(self._wait_process(0, timeout=90), -6)
  File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3721, in assertEqual
    raise error_metas.pop()[0].to_error(
AssertionError: None mismatch: None is not -6
```

**Comment by fduwjj** [comment]
@ZainRizvi this indicates that there is timeout, in the class `NCCLTraceTestDumpOnTimeoutBase`

```
    def _wait_process(self, rank, timeout):
        try:
            self.processes[rank].join(timeout)
            return self.processes[rank].exitcode
        except TimeoutError:
            return None
```

When it's None that means the parent process waited for more than 90 seconds... which is weird, unless the first all-reduce takes more than 90 secs.

**Comment by fduwjj** [comment]
@ZainRizvi what is usual way to debug this? By ssh into the CI machine?

**Comment by ZainRizvi** [comment]
> @ZainRizvi what is usual way to debug this? By ssh into the CI machine?

Yes. You'll want to ssh into the machine while the CI is still running and jump onto the docker container.  

@wconstab mentioned using this command to connect to the container: `docker exec -it $(docker container ps --format '{{.ID}}') bash`

**Review by kit1980 — APPROVED** [review]

**Comment by c-p-i-o** [comment]
Not sure why these set of tests are failing on the new AMI.


> > where can we see the test failure? I can help with debugging these tests.
> 
> @c-p-i-o is looking into the test failure.
> 
> Here's an example of what they look like though:
> 
> Examples of the failures: Failure 1: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963175
> 
> ```
> FAILED [90.0880s] distributed/test_c10d_nccl.py::NCCLTraceTestDumpOnTimeout::test_timeout_dumps_timing_enabled_False - AssertionError: None mismatch: None is not -6
> ```
> 
> Failure 2: https://github.com/pytorch/pytorch/actions/runs/10047579686/job/27770963494
> 
> ```
> ____ NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks ____
> Traceback (most recent call last):
>   File "/var/lib/jenkins/workspace/test/distributed/test_c10d_nccl.py", line 4214, in test_timeout_dumps_on_stuck_ranks
>     self.assertEqual(self._wait_process(0, timeout=90), -6)
>   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3721, in assertEqual
>     raise error_metas.pop()[0].to_error(
> AssertionError: None mismatch: None is not -6
> ```

Not sure why the signal isn't being caugh in the wait program in this new AMI.
Super wierd.
The `_wait_process` function is simple.

**Review by wconstab — APPROVED** [review]
review to unblock the AMI upgrade, as IIUC we have no choice but to upgrade.  That said, we're not sure how to fix this an we're losing test coverage here.  we could use help digging into the actual breakage

**Comment by ZainRizvi** [comment]
@wconstab @c-p-i-o @fduwjj  Moving the conversation to https://github.com/pytorch/pytorch/issues/132428

**Pull request closed without merging**